core/typedefs.h

```diff
@@ -139,9 +139,22 @@ inline void __swap_tmpl(T &x, T &y) {
 }
 #endif // SWAP
 
+#if defined(__GNUC__) || defined(__clang__)
+/* GCC, Clang */
+#if defined(__i386__) || defined(__ILP32__) || (defined(__SIZEOF_POINTER__) && (__SIZEOF_POINTER__ == 4))
+#define IS_32_BIT
+#endif
+#elif defined(_MSC_VER)
+/* Microsoft Visual C++ */
+#if defined(_M_IX86)
+#define IS_32_BIT
+#endif
+#endif
+
 /* Functions to handle powers of 2 and shifting. */
 
 // Function to find the next power of 2 to an integer.
+
```

Comment:
Why? :) The comment directly qualifies the method under it.

Reply:
oops, sorry